crates/fuel-core/src/service.rs

```diff
@@ -251,6 +258,7 @@ impl RunnableTask for Task {
                 );
             }
         }
+        self.shared.database.flush()?;
```

Comment:
Nit: There are a few places that we're calling `self.something.somethig_else...`.

It might be better to create a helper functions for each of these:
`self.flush_database()`, `self.flush_view_layer()`, etc

see: https://en.wikipedia.org/wiki/Law_of_Demeter